homeassistant/components/webdav/camera.py

```diff
@@ -0,0 +1,234 @@
+    vol.Optional(CONF_KEY_PATH): cv.string,
+    vol.Optional(CONF_PATH, default="/"): cv.string,
+    vol.Required(CONF_IMAGE_INTERVAL): cv.time_period,
+})
+
+# This must be separate from frame interval since listing can be expensive
+SCAN_INTERVAL = timedelta(minutes=15)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up a single webdav camera."""
+    from webdav3.client import Client
+    from webdav3.exceptions import WebDavException
+
+    name = config[CONF_NAME]
+    host = config[CONF_HOST]
+    path = config[CONF_PATH]
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    token = config.get(CONF_TOKEN)
+    cert_path = config.get(CONF_CERTIFICATE_PATH)
+    key_path = config.get(CONF_KEY_PATH)
+    client = Client({
+        "webdav_hostname": host,
+        "webdav_login": username,
+        "webdav_password": password,
+        "webdav_token": token,
+        "webdav_cert_path": cert_path,
+        "webdav_key_path": key_path,
+        "webdav_root": path,
+    })
+    session = async_get_clientsession(hass)
+
+    try:
+        client.list('/')  # This will throw if we can't access the share
+        add_entities(
+            [WebDavCamera(name, client, session, config[CONF_IMAGE_INTERVAL])],
+            update_before_add=True)
+    except WebDavException as exception:
+        _LOGGER.warning(
+            "Failed to connect to %s: %s",
+            client.get_url(""),
+            exception)
+        raise PlatformNotReady()
+
+
+class WebDavCamera(Camera):
+    """Models a WebDAV share as a camera.
+
+    Displays image files in the share in sorted order by filename.
+    """
+
+    def __init__(self, name, client, session, image_interval):
+        """Initialize the webdav camera."""
+        super().__init__()
+        self._available = True
+        self._has_images = True
+        self._client = client
+        self._directory = '/'
+        self._files = []
+        self._image_interval = image_interval
+        self._image_number = 0
+        self._image = None
+        self._image_lock = None
+        self._name = name
+        self._session = session
+        self._stop_advancing = None
+
+    async def async_added_to_hass(self):
+        """Set up periodic image advancement."""
+        self._image_lock = asyncio.Lock()
+        self.turn_on()
+        await self._advance()
+
+    @property
+    def stream_source(self):
+        """Return the proxy stream path."""
+        return STREAM_URL.format(self.entity_id, self.access_tokens[-1])
+
+    @property
+    def should_poll(self):
+        """Return True.
+
+        This camera needs to poll because it's pulling from a file share and
+        the contents of the share may change.
+        """
+        return True
+
+    @property
+    def supported_features(self):
+        """Return supported features."""
+        return [SUPPORT_ON_OFF, SUPPORT_STREAM]
```

Comment:
SUPPORT STREAM means you support RTSP. The built in API for streaming is not the same. Please remove this support flag and the `stream_source` property.